assets/components/IconApp.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+    <div>
+        <button :class="['icons', 'cursor-pointer']" :style="{ backgroundColor: color }" @click="handleClick">
+            <font-awesome-icon :icon="iconClass" size="2x" />
+        </button>
+    </div>
+</template>
+
+<script setup lang="ts">
+const props = defineProps({
```

Comment:
Please use the typed variant of defineProps (doc at https://vuejs.org/guide/typescript/composition-api.html#typing-component-props)